### PR TITLE
Create a data factory for settings [MAILPOET-1665]

### PR DESCRIPTION
**Pull request by pavel-mailpoet**

### CODE REVIEW

File: tests/acceptance/EditSignUpConfirmationEmailCest.php [hunk 1]
```diff
@@ -3,21 +3,25 @@
...
     $I->amOnMailPoetPage('Settings');
     $I->click('[data-automation-id="signup_settings_tab"]');
     $I->waitForText('Enable sign-up confirmation');
 
-    // make sure sign up confirmation is enabled
-    $I->click('[data-automation-id="enable_signup_confirmation"]');
-    $I->acceptPopup();
-
     // edit confirmation email
     $I->fillField('[data-automation-id="signup_confirmation_email_from_name"]', 'Confirmation Test From');
```
Comment (amine-mp) [inline tests/acceptance/EditSignUpConfirmationEmailCest.php]: I would suggest adding a method `withConfirmationEmailFrom($name, $email)` to configure the "From name" and "From email".

File: tests/acceptance/EditSignUpConfirmationEmailCest.php [hunk 2]
```diff
@@ -3,21 +3,25 @@
...
     $I->fillField('[data-automation-id="signup_confirmation_email_from_email"]', 'from-confirmation-test@example.com');
```
Comment (amine-mp) [inline tests/acceptance/EditSignUpConfirmationEmailCest.php]: methods `withConfirmationEmailSubject` and `withConfirmationEmailBody` should be used to replace the lines below. No need to fill inputs and submit the form.

**Comment by costasovo** [comment]
I've added `withConfirmationEmailFrom` method which might be useful, but I think it is nor good idea to use methods you suggest within this particular test because the test is actually testing the settings form.

**Comment by amine-mp** [comment]
Yes, you're right, my bad.

**Pull request merged**